slug.go

```diff
@@ -149,14 +149,21 @@ func packWalkFn(root, src, dst string, tarW *tar.Writer, meta *Meta, dereference
 			return nil
 		}
 
-		if m := matchIgnoreRule(subpath, ignoreRules); m {
-			return nil
+		if m, r := matchIgnoreRule(subpath, ignoreRules); m && !(r.val == "**/*" && info.IsDir()) {
```

Comment:
I'm not terribly happy about having to include `**/*` in here, but it makes it possible to write ignorefiles like:
```ignore
.git/
node_modules/
*
!*.tf
!*.json
!*.hcl
```
and get only the `tf`, `json` and `hcl` files. Without this, the `*` matches all directories and prevents recursion into anything that isn't explicitly allowed.